lib/pure/asyncmacro.nim

```diff
@@ -447,8 +325,28 @@ macro multisync*(prc: untyped): untyped =
   result = newStmtList()
   result.add(asyncSingleProc(asyncPrc))
   result.add(sync)
-
-proc await*[T](x: T) =
-  ## The 'await' keyword is also defined here for technical
-  ## reasons. (Generic symbol lookup prepass.)
-  {.error: "Await only available within .async".}
+  # echo result.repr
+
+# overload for type which is not Future[T], based on the yglukhov's patch to chronos: https://github.com/status-im/nim-chronos/pull/47
+template await*(f: untyped): untyped =
+  when declared(retFuture):
+    static:
+      error "await expects Future[T], got " & $typeof(f)
+  elif not declared(inMultisync):
+    static:
+      error "await only available within {.async.}"
+  else:
+    f
+
+# based on the yglukhov's patch to chronos: https://github.com/status-im/nim-chronos/pull/47
+template await*[T](f: Future[T]): auto =
+  when declared(retFuture):
+    when not declared(internalTmpFuture):
+      var internalTmpFuture: FutureBase
+    internalTmpFuture = f
```

Comment:
Why do you need this temporary variable?

Reply:
maybe because of the `FutureBase` type? but this should be cast-able

Reply:
you're right, i need it but not with `when not declared` it seems it doesnt leak now